src/python_testing/TestBatchInvoke.py

```diff
@@ -0,0 +1,111 @@
+#    limitations under the License.
+#
+import logging
+
+import chip.clusters as Clusters
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main, type_matches
+from mobly import asserts
+
+''' Integration test of batch commands using UnitTesting Cluster
+
+This test is meant to test cases not covered in IDM_1_4 as a result of not being able to control
+how DUT processes commands.
+'''
+class TestBatchInvoke(MatterBaseTest):
+
+    @async_test_body
+    async def test_batch_invoke(self):
+        dev_ctrl = self.default_controller
+        dut_node_id = self.dut_node_id
+
+        self.print_step(0, "Commissioning - already done")
+
+        self.print_step(1, "Get remote node's MaxPathsPerInvoke")
+        session_parameters = dev_ctrl.GetRemoteSessionParameters(dut_node_id)
+        max_paths_per_invoke = session_parameters.maxPathsPerInvoke
+
+        asserts.assert_greater_equal(max_paths_per_invoke, 2, "Test expects to be able to send at least 2 commands")
+        asserts.assert_less_equal(max_paths_per_invoke, 65535, "Max Paths per Invoke greater than spec max")
+
+        self.print_step(1, "Send simple batch commands to UnitTesting Cluster")
+        response_size = 7
+        endpoint = 1
+        request_1_fill_character = ord('a')
+        request_2_fill_character = ord('b')
+        command = Clusters.UnitTesting.Commands.TestBatchHelperRequest(
+            sleepBeforeResponseTimeMs=0, sizeOfResponseBuffer=response_size, fillCharacter=request_1_fill_character)
+        invoke_request_1 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+        command = Clusters.UnitTesting.Commands.TestSecondBatchHelperRequest(
+            sleepBeforeResponseTimeMs=0, sizeOfResponseBuffer=response_size, fillCharacter=request_2_fill_character)
+        invoke_request_2 = Clusters.Command.InvokeRequestInfo(endpoint, command)
+        try:
+            result = await dev_ctrl.SendBatchCommands(dut_node_id, [invoke_request_1, invoke_request_2])
+        except InteractionModelError as e:
+            asserts.fail("DUT failed to successfully responded to a InvokeRequest action with two valid commands")
+        
+        asserts.assert_true(type_matches(result, list), "Unexpected return from SendBatchCommands")
+        asserts.assert_equal(len(result), 2, "Unexpected number of InvokeResponses sent back from DUT")
+        asserts.assert_true(type_matches(
+            result[0], Clusters.UnitTesting.Commands.TestBatchHelperResponse), "Unexpected return type for first InvokeRequest")
+        asserts.assert_true(type_matches(
+            result[1], Clusters.UnitTesting.Commands.TestBatchHelperResponse), "Unexpected return type for second InvokeRequest")
+        asserts.assert_equal(len(result[0].buffer), response_size, "Unexpected response size for first InvokeRequest")
+        for byte in result[0].buffer:
```

Comment:
what type is result[0].buffer? if list could we compare as:

```
result[0].buffer == [request_1_fill_character]*response_size
```
and we catch both size and content checks?

Reply:
It is a byte_string, so I don't think we can do that